kliff/models/kim.py

```diff
@@ -602,45 +599,29 @@ def set_one_opt_param(self, name: str, settings: List[List[Any]]):
         # reset influence distance in case it changes
         self.init_influence_distance()
 
-    def update_model_params(self, params: Sequence[float]):
+    def update_model_params(
+        self, params: Union[np.ndarray, List[Union[float, int, Parameter]]]
```

Comment:
The major use case of this function is to update model params from an optimizer. 

I believe for most scipy optimizer, it expects an 1D array as the parameter. As a result, the argument `params` here typically is a 1D array. In this case, seems `super().update_model_params(params)` in line 610 won't work because it expects a list of 1D array (see other comments there)?

Reply:
Updated the `Model.update_model_params` function to update python copy of the function, while the `KIMModel` override also updates the C++ memory bound variables.